Introduce qspi_inst_t type for QSPI instructions

kyle-cypress opened the pull request:
### Description
Encourage the usage of consistent types (there are currently  a mix of `int` and `unsigned int` used for qspi instructions)
QSPI commands are limited to 8 bits, to this is a typdef to uint8_t

<!--
    Required
    Add here detailed changes summary, testing results, dependencies
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html (Pull request template)
-->


### Pull request type

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [ ] Fix
    [x] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers

<!--
    Optional
    Request additional reviewers with @username
-->

### Release Notes

<!--
    Optional
    In case of breaking changes, functionality changes or refactors, please add release notes here. 
    For more information, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html#pull-request-types).
-->

Commit:
Introduce qspi_inst_t type for QSPI instructions

Encourage the usage of consistent types (there are currently
 a mix of `int` and `unsigned int` used for qspi instructions)
QSPI commands are limited to 8 bits, to this is a typdef to char

0xc0170 commented on drivers/QSPI.h:
```diff
@@ -39,6 +41,10 @@ namespace mbed {
 
+/** Type representing a QSPI instruction
+ */
+typedef uint8_t qspi_inst_t;
```
`qspi_status_t` in the HAL defines the instruction as `uint8_t`. However, the drivers use "fundamental types", see other drivers like SPI. See https://os.mbed.com/handbook/C-Data-Types for details.
Although we use C99 types in HAL.

For consistency, I would leave this as it is or at least go with "unsigned char". We could challenge using these fundamental types in drivers, I recall we were asked previously. Like this one: https://os.mbed.com/questions/1749/What-are-the-preferred-data-types-for-mb/

We should fix the storage if they use unsigned for whatever reason to use the same one.

kyle-cypress replied:
Updated the typdef to use `char` instead.

bulislaw reviewed (approved):
Looks good and shouldn't break source backward compatibility.

VeijoPesonen commented on drivers/source/QSPI.cpp:
```diff
@@ -246,12 +246,12 @@ bool QSPI::_acquire()
     //Set up instruction phase parameters
     _qspi_command.instruction.bus_width = _inst_width;
-    if (instruction != -1) {
+    if (instruction != QSPI_NO_INST) {
```
Switch from `-1` to `0` was intentional? Is there some other PR which is related to this change?

kyle-cypress replied:
Since qspi_inst_t is now an unsigned char, -1 is not a valid value. The #define is additionally introduced to improve clarity.
This did however prompt me to check and I noticed that there were two locations where -1 was still being passed to _qspi_build_command; I've pushed up a change updating those to use QSPI_NO_INST.

VeijoPesonen reviewed (approved):
Looks good to me.

0xc0170 committed:
QSPIFBlockDevice: fix type on the namespace mbed

Commit:
QSPIFBlockDevice: Fix incomplete propagation of qspi_inst_t

0xc0170 commented on drivers/QSPI.h:
```diff
@@ -28,6 +28,8 @@
 
 #define ONE_MHZ     1000000
 
+#define QSPI_NO_INST 0x00
```
One additional request, please send new PR: Is this documented anywhere? There are in doxygen notes about: "use this value for this phase to be ignored" . This would be good to have  there as well otherwise this define is not known to a user.

kyle-cypress replied:
Submitted #11691

Commit:
Change qspi_inst_t typdef to int

Existing code may a dependency on the old behavior of "-1" to
mean "no instruction". Therefore, update the typedef, and the value
of QSPI_NO_INST, to avoid breaking those uses.
